ccm_web/server/src/api/api.service.ts

```diff
@@ -47,7 +35,8 @@ export class APIService {
       const course = response.body
       return { id: course.id, name: course.name }
     } catch (error) {
-      return APIService.handleAPIError(error)
+      const errResponse: APIErrorHandler = handleAPIError(error)
```

Comment:
One weird thing i noticed with GetCourseName call is that if we assign `failedInput:undefined` the response object removes it. Why is this?

Reply:
Hm, interesting. It might be because technically any string key not given a non-`undefined` value is `undefined`. I'd have to do some more research on this. It's not concerning to me though.

Reply:
But that kind of make the failedInput optional in the response object, should we make the response object more predictable or send null?

Reply:
I'm okay either way; if you prefer `null`, we can do that, no worries.

Reply:
I made it null. and implemented the API structure as suggested